src/main/java/org/osc/manager/ism/api/IsmDomainApi.java

```diff
@@ -18,35 +18,104 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Callable;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
 
 import org.apache.log4j.Logger;
-import org.osc.manager.ism.model.Domain;
-import org.osc.manager.ism.model.DomainListElement;
+import org.osc.manager.ism.entities.ApplianceManagerConnectorEntity;
+import org.osc.manager.ism.entities.DomainEntity;
 import org.osc.sdk.manager.api.ManagerDomainApi;
 import org.osc.sdk.manager.element.ApplianceManagerConnectorElement;
+import org.osc.sdk.manager.element.ManagerDomainElement;
+import org.osgi.service.transaction.control.TransactionControl;
 
 public class IsmDomainApi implements ManagerDomainApi {
 
     Logger log = Logger.getLogger(IsmDomainApi.class);
+    private TransactionControl txControl;
+    private EntityManager em;
+    ApplianceManagerConnectorElement mc;
+
+    public IsmDomainApi(ApplianceManagerConnectorElement mc, TransactionControl txControl, EntityManager em)
+            throws Exception {
+        this.txControl = txControl;
+        this.em = em;
+        this.mc = mc;
+    }
 
-    public IsmDomainApi(ApplianceManagerConnectorElement mc) throws Exception {
+    public static IsmDomainApi create(ApplianceManagerConnectorElement mc, TransactionControl txControl,
+            EntityManager em) throws Exception {
 
+        return new IsmDomainApi(mc, txControl, em);
     }
 
-    public static IsmDomainApi create(ApplianceManagerConnectorElement mc) throws Exception {
-        return new IsmDomainApi(mc);
+    public String createDomain(ManagerDomainElement dm) throws Exception {
```

Comment:
By the user do you mean OSC? One thing to keep in mind: OSC -> SEC MGR APIs/SDK -> Plugin. So OSC will never call methods in the plugins that are not defined in the SDK/Interfaces. In this case, OSC has no need to create domains through the plugin, just retrieve them. That is the reason why there is no create domain API in the SDK/Interfaces. So yes, creation of domains should be possible only through the rest apis. You can follow this pattern for other APIs/entities of the SDK as well, only operations on the interfaces must be honored by the plugin.